File: backend/tests/order_item_test.rb
require 'test/unit'
require_relative '../src/order_item'

class OrderItemTest < Test::Unit::TestCase

  def setup
    @order_item = OrderItem.new(order: nil, product: nil)
  end

  def test_can_create_order_item
    assert_instance_of(OrderItem, @order_item)
  end

  def test_can_read_item_data
    assert_nil(@order_item.order)
    assert_nil(@order_item.product)
  end

  def test_can_get_total
    assert_equal(10, @order_item.total)
  end
end

File: backend/proposal/core/test/order_item_test.py
import unittest
from core.order_item import OrderItem
from core.product import Product
from core.order import Order
from core.customer import Customer

class TestOrderItem(unittest.TestCase):
    # Testing if Invoice 
    def test_creation(self):
        order = Order(Customer())
        product = Product("test name", "book")
        order_item = OrderItem(order, product)
        self.assertEqual(order_item.order, order)
        self.assertEqual(order_item.product, product)
    
if __name__ == "__main__":
    unittest.main()
